### PR TITLE
Added some more key level generation items.

### PR DESCRIPTION
**Pull request by matt-gardner2**

### CODE REVIEW

File: DataConverter/Catalyst.DataConverter/HierarchicalDataTransformerHelper.cs [hunk 1]
```diff
@@ -32,6 +34,22 @@ namespace DataConverter
     public class HierarchicalDataTransformerHelper : IHierarchicalDataTransformerHelper
     {
 
+        public async Task<JobData> TransformDataAsync(
```
Comment (jmwestbe) [inline DataConverter/Catalyst.DataConverter/HierarchicalDataTransformerHelper.cs]: This will go away eventually, right?

Comment (matt-gardner2) [reply DataConverter/Catalyst.DataConverter/HierarchicalDataTransformerHelper.cs]: Yes.  Should have taken it out before pushing.  Good catch (it's gone now).

File: DataConverter/Catalyst.DataConverter/HierarchicalDataTransformerHelper.cs [hunk 2]
```diff
@@ -224,10 +238,175 @@ public async Task GetChildText(StringBuilder builder, Binding binding, Binding[]
...
+        /// <param name="bindingReference">
+        /// The binding reference.
+        /// </param>
+        /// <param name="isParent">
+        /// The is parent.
+        /// </param>
+        /// <returns>
+        /// The <see cref="Dictionary"/>.
+        /// </returns>
+        private KeyValuePair<string, string> GetKeyLevelColumnLookup(BindingReference bindingReference, bool isParent)
```
Comment (jmwestbe) [inline DataConverter/Catalyst.DataConverter/HierarchicalDataTransformerHelper.cs]: Unused method?

File: DataConverter/Catalyst.DataConverter/HierarchicalDataTransformerHelper.cs [hunk 3]
```diff
@@ -224,10 +238,175 @@ public async Task GetChildText(StringBuilder builder, Binding binding, Binding[]
...
+        /// <param name="replaceValue">
+        /// The replace value.
+        /// </param>
+        /// <param name="depth">
+        /// The depth.
+        /// </param>
+        /// <returns>
+        /// The <see cref="string"/>.
+        /// </returns>
+        private string GetNewSqlString(string originalString, string replaceKey, string replaceValue, int depth)
```
Comment (jmwestbe) [inline DataConverter/Catalyst.DataConverter/HierarchicalDataTransformerHelper.cs]: Unused Method?

File: DataConverter/Catalyst.DataConverter/HierarchicalDataTransformerHelper.cs [hunk 4]
```diff
@@ -224,10 +238,175 @@ public async Task GetChildText(StringBuilder builder, Binding binding, Binding[]
...
+        /// <summary>
+        /// The strip off json junk.
+        /// </summary>
+        /// <param name="withJsonJunk">
+        /// The with json junk.
+        /// </param>
+        /// <returns>
+        /// The <see cref="string"/>.
+        /// </returns>
+        private string StripOffJsonJunk(string withJsonJunk)
```
Comment (jmwestbe) [inline DataConverter/Catalyst.DataConverter/HierarchicalDataTransformerHelper.cs]: Unused method?

**Review by jmwestbe — CHANGES_REQUESTED** [review]
Could use a little cleanup - mostly unused methods, or at least they seem to be unused

**Review by jmwestbe — APPROVED** [review]

**Pull request merged**